src/streaming-ingestion/flink-etl/src/main/java/software/aws/solution/clickstream/flink/StreamingJob.java

```diff
@@ -0,0 +1,419 @@
+                     DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                     String formattedDateTime = dateTime.format(formatter);
+                     data.put("event_date", formattedDateTime);
+
+                     data.set("event_dimensions", null);
+
+                     ArrayNode eventParams = jsonParser.createArrayNode();
+                     ObjectNode eventParam = jsonParser.createObjectNode();
+                     Transformer kvTransformer = new KvTransformer();
+                     Map<String, String> kvParamMap = new HashMap<>();
+                     kvParamMap.put(KvTransformer.PARAM_KEY_NAME, "_session_id");
+                     kvParamMap.put(KvTransformer.PARAM_KEY_STRING_VALUE, attributesNode.get("_session_id").asText());
+                     eventParams.add(kvTransformer.transform(eventParam, kvParamMap));
+
+                     eventParam = jsonParser.createObjectNode();
+                     kvParamMap = new HashMap<>();
+                     kvParamMap.put(KvTransformer.PARAM_KEY_NAME, "_session_start_timestamp");
+                     kvParamMap.put(KvTransformer.PARAM_KEY_INT_VALUE, attributesNode.get("_session_start_timestamp").asText());
+                     eventParams.add(kvTransformer.transform(eventParam, kvParamMap));
+
+                     eventParam = jsonParser.createObjectNode();
+                     kvParamMap = new HashMap<>();
+                     kvParamMap.put(KvTransformer.PARAM_KEY_NAME, "_session_duration");
+                     kvParamMap.put(KvTransformer.PARAM_KEY_INT_VALUE, attributesNode.get("_session_duration").asText());
+                     eventParams.add(kvTransformer.transform(eventParam, kvParamMap));
+
+                     eventParam = jsonParser.createObjectNode();
+                     kvParamMap = new HashMap<>();
+                     kvParamMap.put(KvTransformer.PARAM_KEY_NAME, "_session_number");
+                     kvParamMap.put(KvTransformer.PARAM_KEY_INT_VALUE, attributesNode.get("_session_number").asText());
+                     eventParams.add(kvTransformer.transform(eventParam, kvParamMap));
+
+                     eventParam = jsonParser.createObjectNode();
+                     kvParamMap = new HashMap<>();
+                     kvParamMap.put(KvTransformer.PARAM_KEY_NAME, "_screen_name");
+                     kvParamMap.put(KvTransformer.PARAM_KEY_STRING_VALUE, attributesNode.get("_screen_name").asText());
+                     eventParams.add(kvTransformer.transform(eventParam, kvParamMap));
+
+                     eventParam = jsonParser.createObjectNode();
+                     kvParamMap = new HashMap<>();
+                     kvParamMap.put(KvTransformer.PARAM_KEY_NAME, "_screen_id");
+                     kvParamMap.put(KvTransformer.PARAM_KEY_STRING_VALUE, attributesNode.get("_screen_id").asText());
+                     eventParams.add(kvTransformer.transform(eventParam, kvParamMap));
```

Comment:
We can create a helper function to eliminate duplicate code.

Reply:
done